walk-in-interview/src/main/java/com/google/job/data/JobQuery.java

```diff
@@ -0,0 +1,88 @@
+package com.google.job.data;
+
+/** Class for the job listings query using builder pattern. */
+public final class JobQuery {
+    private int minLimit = 0;
+    private int maxLimit = Integer.MAX_VALUE;
+    private SingaporeRegion region = SingaporeRegion.ENTIRE;
+    private Filter sortBy = Filter.SALARY;
+    private Order order = Order.DESCENDING;
+    private int pageSize = 100;
+    private int pageIndex = 0;
+
+    public JobQuery() {}	
+
+    public JobQuery setMinLimit(int minLimit) {
+        assert minLimit >= 0;
+        this.minLimit = minLimit;
+        return this;
+    }
+
+    public JobQuery setMaxLimit(int maxLimit) {
+        assert maxLimit > 0 && maxLimit >= this.minLimit;
+        this.maxLimit = maxLimit;
+        return this;
+    }
+
+    public JobQuery setRegion(SingaporeRegion region) {
+        this.region = region;
+        return this;
+    }
+
+    public JobQuery setSortBy(Filter sortBy) {
+        this.sortBy = sortBy;
+        return this;
+    }
+
+    public JobQuery setOrder(Order order) {
+        this.order = order;
+        return this;
+    }
+
+    public JobQuery setPageSize(int pageSize) {
+        assert pageSize >= 1;
```

Comment:
i wrote that quickly as pseudocode, you should use proper if+throw IllegalArgumentException :)